lib/angular-http-server.js

```diff
@@ -102,24 +105,64 @@ function requestListener(req, res) {
 
     const safeFullFileName = getFilePathFromUrl(req.url, basePath, { baseHref });
 
-    fs.stat(safeFullFileName, function(err, stats) {
-        var fileBuffer;
+    fs.stat(safeFullFileName, function (err, stats) {
         if (!err && stats.isFile()) {
-            fileBuffer = fs.readFileSync(safeFullFileName);
-            let ct = mime.lookup(safeFullFileName);
+            const ct = mime.lookup(safeFullFileName);
+            const acceptEncoding = req.headers['accept-encoding'] || '';
+
             log(`Sending ${safeFullFileName} with Content-Type ${ct}`);
-            res.writeHead(200, { "Content-Type": ct });
+
+            if (contentIsCompressible(ct, acceptEncoding)) {
+                sendCompressedResponse(res, ct, acceptEncoding, safeFullFileName);
+            } else {
+                const raw = fs.createReadStream(safeFullFileName);
+                res.writeHead(200, {
+                    "Content-Type": ct
+                });
+                raw.pipe(res);
```

Comment:
can you explain what raw.pipe does relative the existing code that uses res.write?

Reply:
@simonh1000, the pipe here allows the file that is being read as a stream to be streamed directly to the response without the entire file being in memory at any point. It is a more efficient approach to serving files, especially if any of the static assets is large.